java/org/apache/catalina/authenticator/BasicAuthenticator.java

```diff
@@ -132,6 +132,10 @@ protected String getAuthMethod() {
         return HttpServletRequest.BASIC_AUTH;
     }
 
+    @Override
+    protected boolean isPreemptiveAuthRequest(Request request) {
+        return request.getCoyoteRequest().getMimeHeaders().getValue("authorization") != null;
```

Comment:
I don't consider this to be preemptive authentication. It says nothing that the use has not been challenged.

Reply:
It's the way that this method is called from `AuthenticatorBase` that makes it a question for preemptive. The call is combined with a check of `Context.getPreemptiveAuthentication()` there. Any suggestion for a better name? How about `checkPreemptiveAuthRequirements()`?

Reply:
I prefer the latter.

Reply:
Renamed method `isPreemptiveAuthRequest()` to `isPreemptiveAuthPossible()`